src/Packetery/Module/Log/Purger.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Class Page
+ *
+ * @package Packetery\Module\Log
+ */
+
+declare( strict_types=1 );
+
+
+namespace Packetery\Module\Log;
+
+/**
+ * Class Page
+ *
+ * @package Packetery\Module\Log
+ */
+class Purger {
+
+	private const CRON_AUTO_DELETION_HOOK = 'packetery_cron_log_auto_deletion_hook';
+
+	/**
+	 * Log repository.
+	 *
+	 * @var Repository
+	 */
+	private $logRepository;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Repository $logRepository Log repository.
+	 */
+	public function __construct( Repository $logRepository ) {
+		$this->logRepository = $logRepository;
+	}
+
+	/**
+	 * Registers purger.
+	 *
+	 * @return void
+	 */
+	public function register(): void {
+		add_action( self::CRON_AUTO_DELETION_HOOK, [ $this, 'autoDeleteHook' ] );
+		if ( ! wp_next_scheduled( self::CRON_AUTO_DELETION_HOOK ) ) {
+			wp_schedule_event( ( new \DateTime( 'next day 02:00', wp_timezone() ) )->getTimestamp(), 'daily', self::CRON_AUTO_DELETION_HOOK );
+		}
+	}
+
+	/**
+	 * Unregisters purger.
+	 *
+	 * @return void
+	 */
+	public static function deactivate(): void {
+		wp_clear_scheduled_hook( self::CRON_AUTO_DELETION_HOOK );
+	}
+
+	/**
+	 * Hook for auto-deletion.
+	 *
+	 * @return void
+	 */
+	public function autoDeleteHook(): void {
+		$this->autoDelete( 90 );
+	}
+
+	/**
+	 * Auto delete.
+	 *
+	 * @param int $maxRecordAgeInDays Max number of days that record can exist.
+	 *
+	 * @return void
+	 */
+	private function autoDelete( int $maxRecordAgeInDays ): void {
+		$result = $this->logRepository->find(
+			[
+				'date_query' => [
+					[
+						'before' => '- ' . $maxRecordAgeInDays . ' days',
+					],
+				],
+			]
+		);
+
+		foreach ( $result as $record ) {
+			$this->logRepository->delete( (int) $record->id );
```

Comment:
Pokud denne pribyva vetsi nez male mnozstvi zaznamu v tabulce s logy (radove desitky, nez jednotky), zvazil bych, zda je radeji nesmazat jednim 
(pseudokod)
deleteMany( array $ids)  = DELETE FROM logs where id IN ($ids) nebo nejak davkove po mensich blokach idecek

Reply:
Jaký je minimální počet znaků MySQL dotazu v nejhorším relevantním případě? Jakou by jsi dal velikost bloku pro INT(11)?